### ISSUE TITLE
Example of how to securely generate the secret key

### ISSUE THREAD

**Issue by izikandrw** [maxdeviant/redux-persist-transform-encrypt#28]
I imagine that many apps using this library would want to generate an encryption key rather than hardcoding it in the code and having an example of this would be great.

I have been combing the Internet to find out how to do this with no luck. I realize this isn't a function of the library but having an example or suggestion of how to generate the secret key would make it easier for people to use it without hardcoding a secret key in their code.

**Comment by robmoorman** [comment]
Use uuid. And save it into your (react-native-)keychain

**Comment by pitops** [comment]
@robmoorman How about a web solution?

**Comment by MNF** [comment]
Some recommendations are in https://github.com/maxdeviant/redux-persist-transform-encrypt/issues/16

**Comment by angus-leung22** [comment]
> Use uuid. And save it into your (react-native-)keychain

Does anyone know if this is possible? I've tried this approach but running into issues as getting the uuid back from react-native-keychain is an async operation. I cannot wait on it to finish retrieving as I need the value provide it in the transform as the secret key. 

Is there a way to wait on the key prior to the redux store creation or when this transform is called?

**Comment by danwhite-ipc** [comment]
@angus-leung22 Did you manage to find a work around?

**Comment by angus-leung22** [comment]
Unfortunately nothing for getting redux-persist-transform-encrypt to work for me.

**Comment by maxdeviant** [comment]
> Is there a way to wait on the key prior to the redux store creation or when this transform is called?

This would be outside the scope of this library.

It may be possible if you defer the creation of the Redux store (e.g., using a top-level Promise), but that sounds like it could get pretty messy.

**Comment by maxdeviant** [comment]
Refer to the ["Secret Key Selection" section](https://github.com/maxdeviant/redux-persist-transform-encrypt#secret-key-selection) in the README for some recommendations on how to generate a secret key.

I don't use this library (or even Redux) anymore, so I will leave it to someone else to contribute a full-blown example. I would prefer it if an example on key generation was in the form of a blog post or an example repo that we can just link to from the README.

**Comment by lukeharbord** [comment]
@ou2s I'm stuck on this exact issue. Most options for getting a secret key seem to be async - have you got any snippets of how you created the store asynchronously and deferred rendering the Provider? Any help appreciated!

**Comment by asdub** [comment]
@ou2s 
I'm also trying to figure this out. Any advice would be appreciated.

**Comment by ou2s** [comment]
@asdub I think I've managed, but at the end I didn't, there were some defects. I don't think that you can create a redux store asynchronously. I ended up using mkkv-storage library which has synchronous native code bindings thanks to JSI.
